addons/calendar/models/calendar.py

```diff
@@ -997,9 +997,8 @@ def get_search_fields(self, order_fields, r_date=None):
             else:
                 sort_fields[field] = self[field]
                 if isinstance(self[field], models.BaseModel):
-                    name_get = self[field].name_get()
-                    if len(name_get) and len(name_get[0]) >= 2:
-                        sort_fields[field] = name_get[0][1]
+                    name_get = self[field].mapped('display_name')
+                    sort_fields[field] = name_get and name_get[0] or ''
```

Comment:
You are changing a `name_get` to a `display_name`, which looks too much like a refactoring to me.
I suggest instead to add an `else` clause to the original `if`:
```python
else:
    sort_fields[field] = ""
```

Reply:
@rco-odoo actually it is a cherry pick on a fix I made in 10.0 at #26908
As the display_name and name_get are supposed to be the same (and simplified the code), I assumed it was ok